src/main/java/io/kestra/plugin/surrealdb/Query.java

```diff
@@ -0,0 +1,127 @@
+package io.kestra.plugin.surrealdb;
+
+import com.surrealdb.driver.SyncSurrealDriver;
+import com.surrealdb.driver.model.QueryResult;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.models.tasks.common.FetchType;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.FileSerde;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.io.*;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(title = "Query a Surreal database with SurrealQL.")
+@Plugin(
+	examples = {
+		@Example(
+			title = "Send a SurrealQL query to a Surreal database",
+			code = {
+				"useTls: true",
+				"port: 8000",
+				"host: localhost",
+				"username: surreal_user",
+				"password: surreal_passwd",
+				"database: surreal_db",
+				"namespace: surreal_namespace",
+				"query: SELECT * FROM SURREAL_TABLE",
+				"fetchType: STORE"
+			}
+		)
+	}
+)
+public class Query extends SurrealDBConnection implements RunnableTask<Query.Output>, QueryInterface {
+
+	@NotNull
+	@Builder.Default
+	protected FetchType fetchType = FetchType.STORE;
+
+	@Builder.Default
+	protected Map<String, String> parameters = new HashMap<>();
+
+	@NotNull
+	@NotBlank
+	protected String query;
+
+	@Override
+	public Query.Output run(RunContext runContext) throws Exception {
+		SyncSurrealDriver driver = super.connect(runContext);
+
+		String renderedQuery = runContext.render(query);
+
+		List<QueryResult<Object>> results = driver.query(renderedQuery, parameters, Object.class);
+
+		Query.Output.OutputBuilder outputBuilder = Output.builder().size(results.stream()
+			.mapToLong(result -> (long) result.getResult().size())
+			.sum());
+
+		super.disconnect();
+
+		return (switch (fetchType) {
+			case FETCH -> outputBuilder.rows(getResultStream(results).toList());
+			case FETCH_ONE -> outputBuilder.row(getResultStream(results).findFirst().orElse(null));
+			case STORE -> outputBuilder.uri(getTempFile(runContext, getResultStream(results).toList()));
+			default -> outputBuilder;
+		}).build();
+	}
+
+	private static Stream<Map<String, Object>> getResultStream(List<QueryResult<Object>> results) {
+		return results.stream()
+			.map(QueryResult::getResult)
+			.flatMap(list -> list.stream().map(object -> (Map<String, Object>) object));
+	}
+
+	private static URI getTempFile(RunContext runContext, List<Map<String, Object>> results) throws IOException {
+		File tempFile = runContext.tempFile(".ion").toFile();
+		BufferedWriter fileWriter = new BufferedWriter(new FileWriter(tempFile));
```

Comment:
Please add the BufferedWritter inside the try-with-resource to avoid potential resource leaks